apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/peer/ShardDownloader.java

```diff
@@ -183,14 +189,14 @@ private FileDownloadDecision checkShard(Long shardId, Set<Peer> shardPeers) {
         }
         PeerValidityDecisionMaker pvdm = new PeerValidityDecisionMaker(shardPeerList);
         FileDownloadDecision res = pvdm.calcualteNetworkState();
-        goodPeers = pvdm.getValidPeers();
-        badPeers = pvdm.getInvalidPeers();
-        log.debug("prepareForDownloading(), res = {}, goodPeers = {}, badPeers = {}", res, goodPeers, badPeers);
+        goodPeersMap.put(shardId,pvdm.getValidPeers());
+        badPeersMap.put(shardId,pvdm.getInvalidPeers());
+        log.debug("prepareForDownloading(), res = {}, goodPeers = {}, badPeers = {}", res, goodPeersMap.get(shardId), badPeersMap.get(shardId));
```

Comment:
Pls add 'toString()' implementation into classes - PeerFileInfo, PeerShardInfo
otherwise that log will not show correct information.